django_celery_monitor/admin.py

```diff
@@ -78,18 +79,18 @@ def tstamp(task):
     it as a "natural date" -- a human readable version.
     """
     value = make_aware(task.tstamp)
-    return '<div title="{0}">{1}</div>'.format(
-        escape(str(value)), escape(naturaldate(value)),
-    )
+    return format_html('<div title="{0}">{1}</div>', str(value),
+                       naturaldate(value))
 
 
 @display_field(_('name'), 'name')
 def name(task):
     """Return the task name and abbreviates it to maximum of 16 characters."""
-    short_name = abbrtask(task.name, 16)
-    return '<div title="{0}"><b>{1}</b></div>'.format(
-        escape(task.name), escape(short_name),
-    )
+    short_name = task.name
+    if task.name and len(task.name) > 16:
+        short_name = format_html('{0}&hellip;', task.name[0:15])
```

Comment:
Please continue to use `abbrtask` here since it does more than what this code does. Just cutting off the task name would make identifiying a task with a long import path hard to read.

E.g. Imagine a task with the name `myproject.some_complex_app.under_module.tasks.version_2.whatever_task` (something that I've seen myself). In that case I wouldn't even see the task name with this new code. Instead `abbrtask` will show `[.]whatever_task`.

Reply:
Although I'm sympathetic to your concerns about the method in which it abbreviates, sticking with `abbrtask` from Celery simply didn't seem like an option.

The `abbrtask` function from Celery doesn't actually do what the docstring of the `name` function promises. It doesn't cut off at the requested amount of characters. So when I wrote unit tests for the `name` function, I found behavior that contradicts promises made in the docstring.

I had to choose between:

- Rewriting the failing test to accept incorrect behavior
- Sticking with the `abbrtask` function but doing postprocessing on it as a workaround
- Stop using the `abbrtask` function

I went for option 3. Would you make a different choice?